dmodel/dm-macros.h

```diff
@@ -0,0 +1,144 @@
+ *
+ * For development cycles, this means the next stable cycle.
+ */
+#if (DM_MINOR_VERSION == 99)
+#define DM_VERSION_CUR_STABLE (G_ENCODE_VERSION (DM_MAJOR_VERSION + 1, 0))
+#elif (DM_MINOR_VERSION % 2)
+#define DM_VERSION_CUR_STABLE (G_ENCODE_VERSION (DM_MAJOR_VERSION, DM_MINOR_VERSION + 1))
+#else
+#define DM_VERSION_CUR_STABLE (G_ENCODE_VERSION (DM_MAJOR_VERSION, DM_MINOR_VERSION))
+#endif
+
+/**
+ * DM_VERSION_PREV_STABLE:
+ *
+ * Evaluates to the previous stable version.
+ */
+#if (DM_MINOR_VERSION == 99)
+#define DM_VERSION_PREV_STABLE (G_ENCODE_VERSION (DM_MAJOR_VERSION + 1, 0))
+#elif (DM_MINOR_VERSION % 2)
+#define DM_VERSION_PREV_STABLE (G_ENCODE_VERSION (DM_MAJOR_VERSION, DM_MINOR_VERSION - 1))
+#else
+#define DM_VERSION_PREV_STABLE (G_ENCODE_VERSION (DM_MAJOR_VERSION, DM_MINOR_VERSION - 2))
+#endif
+
+/**
+ * DM_VERSION_MIN_REQUIRED:
+ *
+ * A pre-processor symbol that should be defined by the user prior to including
+ * the `dmodel.h` header.
+ *
+ * The value should be one of the predefined DModel version macros,
+ * for instance: %DM_VERSION_0_0, %DM_VERSION_0_2, ...
+ *
+ * This symbol defines the earliest version of DModel that a project
+ * is required to be able to compile against.
+ *
+ * If the compiler is configured to warn about the use of deprecated symbols,
+ * then using symbols that were deprecated in version %DM_VERSION_MIN_REQUIRED,
+ * or earlier versions, will emit a deprecation warning, but using functions
+ * deprecated in later versions will not.
+ *
+ * See also: %DM_VERSION_MAX_ALLOWED
+ */
+#ifndef DM_VERSION_MIN_REQUIRED
+# define DM_VERSION_MIN_REQUIRED (DM_VERSION_CUR_STABLE)
+#endif
+
+/**
+ * DM_VERSION_MAX_ALLOWED:
+ *
+ * A pre-processor symbol that should be defined by the user prior to including
+ * the `dmodel.h` header.
+ *
+ * The value should be one of the predefined DModel version macros,
+ * for instance: %DM_VERSION_0_0, %DM_VERSION_0_2, ...
+ *
+ * This symbol defines the earliest version of DModel that a project
+ * is required to be able to compile against.
+ *
+ * If the compiler is configured to warn about the use of deprecated symbols,
+ * then using symbols that were deprecated in version %DM_VERSION_MAX_ALLOWED,
+ * or later versions, will emit a deprecation warning, but using functions
+ * deprecated in later versions will not.
+ *
+ * See also: %DM_VERSION_MIN_REQUIRED
+ */
+#ifndef DM_VERSION_MAX_ALLOWED
+# if DM_VERSION_MIN_REQUIRED > DM_VERSION_PREV_STABLE
+#  define DM_VERSION_MAX_ALLOWED (DM_VERSION_MIN_REQUIRED)
+# else
+#  define DM_VERSION_MAX_ALLOWED (DM_VERSION_CUR_STABLE)
+# endif
+#endif
+
+/* sanity checks */
+#if DM_VERSION_MAX_ALLOWED < DM_VERSION_MIN_REQUIRED
+#error "DM_VERSION_MAX_ALLOWED must be >= DM_VERSION_MIN_REQUIRED"
+#endif
+#if DM_VERSION_MIN_REQUIRED < DM_VERSION_1_0
+#error "DM_VERSION_MIN_REQUIRED must be >= DM_VERSION_1_0"
+#endif
+
+/* Every new stable minor release should add a set of macros here */
+
+#if DM_VERSION_MIN_REQUIRED >= DM_VERSION_0_0
```

Comment:
We cannot have deprecated API, at this point, so this whole section can be omitted.